In [45]:
import os, openpyxl
import pyperclip as pc
import pandas as pd

### 1. Data import

In [2]:
path = './SourceData/OpenCell/science.abi6983_table_s6.xlsx'

In [30]:
wb = openpyxl.load_workbook(path)
wb.sheetnames

['read_me',
 'localization_annotations',
 'localization_annotations_flatte',
 'Fig S7B']

In [31]:
ws = wb['localization_annotations_flatte']
df = pd.DataFrame(ws.values)
df.head()

0                1                2                 3
0  target_name          ensg_id  annotation_name  annotation_grade
1         AAMP  ENSG00000127837      cytoplasmic                 3
2         AAMP  ENSG00000127837      nucleoplasm                 1
3         AATF  ENSG00000275700     nucleolus_gc                 3
4        ABCE1  ENSG00000164163      cytoplasmic                 3

In [32]:
columns = df.iloc[0,[0,1,2,3]]
print(columns)

0         target_name
1             ensg_id
2     annotation_name
3    annotation_grade
Name: 0, dtype: object


In [35]:
df.columns = columns
df = df.iloc[1:, :]

In [36]:
# df.head(n=12)
df.tail()

0    target_name          ensg_id annotation_name annotation_grade
2745      ZNF326  ENSG00000162664    nucleolus_gc                3
2746      ZNF326  ENSG00000162664     nucleoplasm                3
2747      ZNF598  ENSG00000167962     cytoplasmic                3
2748      ZNF598  ENSG00000167962              er                1
2749         ZYX  ENSG00000159840    cell_contact                3

In [34]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2749 entries, 1 to 2749
Data columns (total 4 columns):
 #   Column            Non-Null Count  Dtype 
---  ------            --------------  ----- 
 0   target_name       2749 non-null   object
 1   ensg_id           2749 non-null   object
 2   annotation_name   2749 non-null   object
 3   annotation_grade  2749 non-null   object
dtypes: object(4)
memory usage: 86.0+ KB


### 2. ID conversion from Ensembl to UniprotKB

In [37]:
ensg_id_list = df['ensg_id'].unique()
len(ensg_id_list)

1310

In [40]:
ensg_id_list_spaced = ' '.join(ensg_id_list)

In [46]:
pc.copy(ensg_id_list_spaced)

#### Paste to ID mapping on Uniprot, filter for Reviewed, then excel output downloaded

#### All of 1310 genes were converted to 1312 IDs

In [49]:
# import of the output excel
df_id_converted = pd.read_excel('./SourceData/OpenCell/Intermediate/Ensg_id_to_uniprot_id.xlsx')
df_id_converted.info()

/Users/ShokenLEE/opt/anaconda3/lib/python3.8/site-packages/openpyxl/styles/stylesheet.py:221: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1312 entries, 0 to 1311
Data columns (total 7 columns):
 #   Column                                                     Non-Null Count  Dtype 
---  ------                                                     --------------  ----- 
 0   yourlist:M202203266320BA52A5CE8FCD097CB85A53697A35478814R  1312 non-null   object
 1   isomap:M202203266320BA52A5CE8FCD097CB85A53697A35478814R    685 non-null    object
 2   Entry                                                      1312 non-null   object
 3   Entry name                                                 1312 non-null   object
 4   Protein names                                              1312 non-null   object
 5   Gene names                                                 1312 non-null   object
 6   Organism                                                   1312 non-null   object
dtypes: object(7)
memory usage: 71.9+ KB


In [50]:
df_id_converted = df_id_converted.drop(['isomap:M202203266320BA52A5CE8FCD097CB85A53697A35478814R'], axis=1)
df_id_converted = df_id_converted.rename(columns={'yourlist:M202203266320BA52A5CE8FCD097CB85A53697A35478814R': 'ensg_id'})
df_id_converted.head()

ensg_id   Entry   Entry name  \
0  ENSG00000127837  Q13685   AAMP_HUMAN   
1  ENSG00000275700  Q9NY61   AATF_HUMAN   
2  ENSG00000164163  P61221  ABCE1_HUMAN   
3  ENSG00000159842  Q12979    ABR_HUMAN   
4  ENSG00000114331  Q15057  ACAP2_HUMAN   

                                       Protein names  \
0            Angio-associated migratory cell protein   
1  Protein AATF (Apoptosis-antagonizing transcrip...   
2  ATP-binding cassette sub-family E member 1 (2'...   
3   Active breakpoint cluster region-related protein   
4  Arf-GAP with coiled-coil, ANK repeat and PH do...   

                                    Gene names              Organism  
0                                         AAMP  Homo sapiens (Human)  
1                        AATF CHE1 DED HSPC277  Homo sapiens (Human)  
2  ABCE1 RLI RNASEL1 RNASELI RNS4I OK/SW-cl.40  Homo sapiens (Human)  
3                                          ABR  Homo sapiens (Human)  
4                        ACAP2 CENTB2 KIAA0041  Homo sapiens (Human)